src/main/java/io/tiledb/spark/Range.java

```diff
@@ -595,4 +595,63 @@ else if (leftOvers > 0) {
 
     return ranges;
   }
+
+  /**
+   * Split a range. Will split upto N buckets
+   *
+   * @param partitions The number of partitions
+   * @param partitionWidth The number of elements in the partition
+   * @return The list of ranges, one per partition
+   * @throws TileDBError
+   */
+  public List<Range> splitRangeToPartitions(Object partitions, Object partitionWidth)
+      throws TileDBError {
+    List<Range> ranges = new ArrayList<>();
+    // Number of buckets is 1 more thank number of splits (i.e. split 1 time into two buckets)
+    // Only long dimensions can be split with naive algorithm
+    Object min = range.getFirst();
+    Object max = range.getSecond();
+    Object currentMin;
+    Object currentMax;
+
+    // If the min is the max this range is not splittable
+    if (!this.splittable()) {
+      ranges.add(new Range(range));
+      return ranges;
+    }
+
+    Object one = null;
+    if (dataClassType == Byte.class) {
+      one = (byte) 1;
+    } else if (dataClassType == Short.class) {
+      one = (short) 1;
+    } else if (dataClassType == Integer.class) {
+      one = 1;
+    } else if (dataClassType == Long.class) {
+      one = 1L;
+    } else if (dataClassType == Float.class) {
+      one = 1f;
+    } else if (dataClassType == Double.class) {
+      one = 1d;
+    }
+
+    currentMin = min;
+    currentMax =
+        util.subtractObjects(
+            util.addObjects(currentMin, partitionWidth, dataClassType), one, dataClassType);
+    ranges.add(new Range(new Pair(currentMin, currentMax)));
+
+    for (int i = 1; i < ((Number) partitions).intValue(); ++i) {
```

Comment:
What happens in this for loop if:

A) you hit `max` before you run out of partitions?
B) You run out of partitions before you hit max?

It seems like in case A) you'll keep building ranges but they will be invalid because they will have ranges of `[max+1, max]`?. For B) It looks like the end of the ranges would be lost?

Reply:
Case A is guarded by https://github.com/TileDB-Inc/TileDB-Spark/blob/118140006c6239f00c34516c9cf7ee98a43ba939/src/main/java/io/tiledb/spark/SubArrayRanges.java#L272

We first check if the partition width based on the `partition_count` is less than 1, and if so, we set the partition width to 1, and the number of partitions to `width`, that is, `max-min` point queries. 

However, to prevent from possible issues related to that, I am adding one extra check:

```
// Check if we reach max before we run out of partitions
      if (util.greaterThanOrEqual(currentMax, max, dataClassType)) {
          currentMax = max;
          ranges.add(new Range(new Pair(currentMin, currentMax)));
          break;
      }
```

Reply:
In case B, the last partition takes the rest of the remaining rows, which usually are a small remainder from the division

Reply:
For B this is only true if the values passed  `partitionWidth*partitions >= max`. If `partitionWidth*partitions < max` you will miss records at the end wont you? You don't have a final check here to say if it's the last partition always set `max` as the `current_max`.

Reply:
ok, added one more check